javaslang/src/test/java/javaslang/LazyTest.java

```diff
@@ -184,22 +190,39 @@ public void shouldSupportMultithreading() {
             Try.run(() -> Thread.sleep(100));
             new Thread(() -> {
                 Try.run(() -> Thread.sleep(100));
-                lock[0] = false;
+                lock.set(false);
             }).start();
-            assertThat(lazy.isEvaluated()).isFalse();
+            isEvaluated.compareAndSet(false, lazy.isEvaluated());
             lazy.get();
         }).start();
+        assertThat(isEvaluated.get()).isFalse();
         assertThat(lazy.get()).isEqualTo(1);
     }
 
+    @Test
+    public void shouldBeConsistentFromMultipleThreads() throws Exception {
+        CompletableFuture.allOf(
```

Comment:
@danieldietrich, I couldn't manage to make it work with slang `Future` (that fails for non-double check), only with `CompletableFuture`